app/views/browse/_replica-set-details.html

```diff
@@ -89,10 +88,18 @@
         add-health-check-url="{{((!deploymentConfig || isActive) && ('deploymentconfigs' | canI : 'update')) ? healthCheckURL : ''}}">
       </pod-template>
       <h4>Volumes</h4>
-      <p ng-if="!deployment.spec.template.spec.volumes.length">
-        <a ng-if="'replicationcontrollers' | canI : 'update'" ng-href="project/{{project.metadata.name}}/attach-pvc?deployment={{deployment.metadata.name}}">Attach storage</a>
-        <span ng-if="!('replicationcontrollers' | canI : 'update')">none</span>
-      </p>
+      <div ng-if="!deployment.spec.template.spec.volumes.length">
+        <div ng-if="kind === 'ReplicaSet'">
+          <a ng-if="{ group: 'extensions', resource: 'replicasets' } | canI : 'update'"
+             ng-href="project/{{project.metadata.name}}/attach-pvc?kind=ReplicaSet&name={{deployment.metadata.name}}&group=extensions">Attach storage</a>
+          <span ng-if="!({ group: 'extensions', resource: 'replicasets' } | canI : 'update')">none</span>
```

Comment:
there are several places here both for RS and RC where you could use resource instead